homeassistant/components/xiaomi_miio/config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Config flow to configure Xiaomi Miio."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_TOKEN
+
+# pylint: disable=unused-import
+from .const import DOMAIN
+from .gateway import ConnectXiaomiGateway
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FLOW_TYPE = "config_flow_device"
+CONF_GATEWAY = "gateway"
+DEFAULT_GATEWAY_NAME = "Xiaomi Gateway"
+
+GATEWAY_CONFIG = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_TOKEN): vol.All(str, vol.Length(min=32, max=32)),
+        vol.Optional(CONF_NAME, default=DEFAULT_GATEWAY_NAME): str,
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema({vol.Optional(CONF_GATEWAY, default=False): bool})
+
+
+class XiaomiMiioFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Xiaomi Miio config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            # Check which device needs to be connected.
+            if user_input.get(CONF_GATEWAY):
```

Comment:
Please go through all the code and remove dict.get where we don't need it. There's a default for this item in the config schema.